apis/database/v1alpha2/sql_types.go

```diff
@@ -260,6 +311,26 @@ type VirtualNetworkRuleStatus struct {
 	Type string `json:"type,omitempty"`
 }
 
+// A PostgresqlVirtualNetworkRuleSpec defines the desired state of a PostgresqlVirtualNetworkRule.
+type PostgresqlVirtualNetworkRuleSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+
+	// Name - Name of the Virtual Network Rule.
+	Name string `json:"name"`
+
+	// ServerName - Name of the Virtual Network Rule's PostgresqlServer.
+	ServerName string `json:"serverName,omitempty"`
+
+	// ServerNameRef - A reference to the Virtual Network Rule's PostgresqlServer.
+	ServerNameRef *PostgresqlServerNameReferencer `json:"serverNameRef,omitempty" resource:"attributereferencer"`
+
+	// ResourceGroupName - Name of the Virtual Network Rule's resource group.
+	ResourceGroupName string `json:"resourceGroupName"`
```

Comment:
Do we want a `ResourceGroupNameRef` in addition to `ResourceGroupName`?

Reply:
Good catch - I imagine we do.

Reply:
fixed in #39